File: app/src/main/java/com/androidevlinux/percy/UTXO/MyApp.java
package com.androidevlinux.percy.UTXO;

import android.app.Application;

import com.androidevlinux.percy.UTXO.utils.ConnectionReceiver;

public class MyApp extends Application {

    private static MyApp mInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
    }

    public static synchronized MyApp getInstance() {
        return mInstance;
    }

    public void setConnectionListener(ConnectionReceiver.ConnectionReceiverListener listener) {
        ConnectionReceiver.connectionReceiverListener = listener;
    }
}

File: app/src/main/java/com/androidevlinux/percy/UTXO/MyApp.kt
package com.androidevlinux.percy.UTXO

import android.app.Application

import com.androidevlinux.percy.UTXO.utils.ConnectionReceiver

class MyApp : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    fun setConnectionListener(listener: ConnectionReceiver.ConnectionReceiverListener) {
        ConnectionReceiver.connectionReceiverListener = listener
    }

    companion object {

        @get:Synchronized
        var instance: MyApp? = null
            private set
    }
}
